erpnext/selling/doctype/customer/customer.py

```diff
@@ -186,6 +188,20 @@ def get_customer_outstanding(customer, company):
 
 	outstanding_based_on_so = flt(outstanding_based_on_so[0][0]) if outstanding_based_on_so else 0.0
 
+	#PR : 10861, Author : ashish-greycube & jigneshpshah,  Email:mr.ashish.shah@gmail.com 
+	# Since the credit limit check is bypassed at sales order level, when customer credit balance report is run we need to treat sales order with status 'To Deliver and Bill' as not outstanding
+	outstanding_based_on_bypassed_so = 0.0
+	bypass_credit_limit_check_at_sales_order = cint(frappe.db.get_value("Customer", customer, "bypass_credit_limit_check_at_sales_order"))
+	if bypass_credit_limit_check_at_sales_order == 1 and caller=='customer_credit_balance_report':
+		outstanding_based_on_bypassed_so = frappe.db.sql("""
```

Comment:
Why this query needed?
We can skip `outstanding_based_on_so` optionally, like in case of report.

Reply:
Nabin, there are two things (1) outstanding_based_on_so and (2) outstanding_based_on_bypassed_so [where credit limit is overruled].
When customer_credit_balance_report is run we should return correct outstanding amount by checking each customer for flag "by pass customer credit limit".
We need to treat sales order with status 'To Deliver and Bill'  and where no advance is paid as NOT outstanding
Hence the logic is "outstanding for by pass credit limit" = "outstanding_based_on_so" ---- (minus) "outstanding_based_on_bypassed_so"